### ISSUE TITLE
Issues with celery and workers - Documentation of configuration required.

### ISSUE THREAD

**Issue by gusmith** [data61/anonlink-entity-service#410]
While running 100 times 1M*1M linkage, I can observe three different errors in the workers:
```
[2019-08-26 06:50:42,148: ERROR/ForkPoolWorker-5] Chord callback for '71e890f0-1bcb-40df-8c9b-d58a7be38da4' raised: ChordError("Dependency de05c27e-a0af-4680-9684-359b868a5faa raised WorkerLostError('Worker exited prematurely: signal 9 (SIGKILL).')")
Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/celery/backends/redis.py", line 399, in on_chord_part_return
    callback.delay([unpack(tup, decode) for tup in resl])
  File "/usr/lib/python3.7/site-packages/celery/backends/redis.py", line 399, in <listcomp>
    callback.delay([unpack(tup, decode) for tup in resl])
  File "/usr/lib/python3.7/site-packages/celery/backends/redis.py", line 352, in _unpack_chord_result
    raise ChordError('Dependency {0} raised {1!r}'.format(tid, retval))
celery.exceptions.ChordError: Dependency de05c27e-a0af-4680-9684-359b868a5faa raised WorkerLostError('Worker exited prematurely: signal 9 (SIGKILL).')
```
after which the `ForkPoolWorker-5` is still working.

In an other worker:
```
[2019-08-26 06:35:55,205: ERROR/MainProcess] Process 'ForkPoolWorker-5' pid:13 exited with 'signal 9 (SIGKILL)'
[2019-08-26 06:35:55,222: ERROR/MainProcess] Task handler raised error: WorkerLostError('Worker exited prematurely: signal 9 (SIGKILL).')
Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/billiard/pool.py", line 1267, in mark_as_worker_lost
    human_status(exitcode)),
billiard.exceptions.WorkerLostError: Worker exited prematurely: signal 9 (SIGKILL).
```
after which the `ForkPoolWorker-5` is not working (surprising it is the same number...)

and finally on the same worker as the previous one, I get a large number of times:
```
[2019-08-26 06:35:59,657: WARNING/ForkPoolWorker-9] /usr/lib/python3.7/site-packages/celery/app/trace.py:568: RuntimeWarning: Exception raised outside body: RecursionError('maximum recursion depth exceeded'):
Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/celery/app/trace.py", line 449, in trace_task
    uuid, retval, task_request, publish_result,
  File "/usr/lib/python3.7/site-packages/celery/backends/base.py", line 149, in mark_as_done
    self.store_result(task_id, result, state, request=request)
  File "/usr/lib/python3.7/site-packages/celery/backends/base.py", line 342, in store_result
    request=request, **kwargs)
  File "/usr/lib/python3.7/site-packages/celery/backends/base.py", line 714, in _store_result
    self.set(self.get_key_for_task(task_id), self.encode(meta))
  File "/usr/lib/python3.7/site-packages/celery/backends/redis.py", line 318, in set
    return self.ensure(self._set, (key, value), **retry_policy)
  File "/usr/lib/python3.7/site-packages/celery/backends/redis.py", line 308, in ensure
    **retry_policy)
  File "/usr/lib/python3.7/site-packages/kombu/utils/functional.py", line 343, in retry_over_time
    return fun(*args, **kwargs)
  File "/usr/lib/python3.7/site-packages/celery/backends/redis.py", line 327, in _set
    pipe.execute()
  File "/usr/lib/python3.7/site-packages/redis/client.py", line 3514, in execute
    self.shard_hint)
  File "/usr/lib/python3.7/site-packages/redis/connection.py", line 986, in get_connection
    self._checkpid()
  File "/usr/lib/python3.7/site-packages/redis/sentinel.py", line 130, in _checkpid
    self.disconnect()
[...]
```
The last four lines (i.e. connection.py and sentinel.py) being repeated a *LARGE* number of times.

**Comment by gusmith** [comment]
The two first issues seem to have been solved by limiting resources available to celery and configuring it well in deployment.

For the last log, I currently have some assumptions: the method `on_chord_error` was able to raise some exceptions if the run was absent, which may be able to call itself (to be confirmed). so instead, catch the exception and log an error.
Otherwise, we also had the task `aggregate_comparisons` which could be retried if we were observing a minio exception, but there was no limit on the number of retries.
These changes are on the branch `fix-on-error-chord`, currently deployed on the benchmark cluster and running.

**Comment by gusmith** [comment]
The two first logged error are now understood, and need documentation to close them.
The last one has now its own issue (issue #412 )
this issue can be closed when documentation about the deployment is provided.

**Comment by gusmith** [comment]
Will be closed by #427

**Comment by gusmith** [comment]
PR #427 merged, documenting the required configuration.
